test/util/testnode/rpc_client.go

```diff
@@ -71,9 +72,17 @@ func StartGRPCServer(logger log.Logger, app srvtypes.Application, appCfg *srvcon
 
 	go blockAPI.StartNewBlockEventListener(cctx.goContext) //nolint:errcheck
 
+	// Create a goroutine-safe logger for the gRPC server to prevent
+	// "Log in goroutine after test has completed" panics when using test loggers
+	grpcLogger := log.NewNopLogger()
+	if logger != nil {
+		// Use a simple stdout logger that won't become invalid when tests complete
+		grpcLogger = log.NewLogger(os.Stdout)
+	}
```

Comment:
[question] why does this modify grpcLogger if logger is not nil?

Reply:
What if logger was configured to log to a file instead of to stdout. Why is it safe to assume the caller wants to log to stdout?